visivo/models/model.py

```diff
@@ -26,3 +44,21 @@ def child_items(self):
             return [self.target]
         else:
             return [DefaultTarget()]
+
+
+Models = Union[SqlModel, ScriptModel]
+
+
+class JoinModel(Model, NamedModel, ParentModel):
```

Comment:
Rather than having a separate join class I was thinking we could just add an attribute to the `sqlModel` & `ScriptModel` classes that tells visivo to store their output in memory maybe looks something like this- 

```
class SqlModel(Model, ParentModel, NamedModel):
    """
    Model that executes sql against a given database target
    """
class SqlModel(Model, ParentModel, NamedModel):
    """
    Model that executes sql against a given database target
    """

    sql: str = Field(
        None,
        description="The sql used to generate your base data",
    )
    store-output: Optional[Boolean] = Field(
       false, 
       description= "Stores the output of your model in the target cached sqlite database" 
    )
    target: Optional[generate_ref_field(Target)] = Field(
        None,
        description="A target object defined inline or a ref() to a chart. Override the defaults.target_name",
    )
    def child_items(self):
        if self.target:
            return [self.target]
        else:
            return [DefaultTarget()]
```

Then since it's stored in a sqlite database the user can write normal sql base models agains the sqlite database to join the items together.

Reply:
One of the issues we are going to run into order of operations.  If you have a model that uses other models, now they have to run in a specific order. 
 
 I was having a thought that maybe these should be targets instead.  I think it might simplify things, I am going to give that a shot.

Reply:
```
models: 
  - name: Model Name
     sql: select * from other_model join third_model on ...
     sub_models:
        - ref(Other Model)
        - ref(Third Model)
  - name: Other Model
     sql: select * from table
  - name: Third Model
     sql: select * from table
```

Reply:
Yup, we'll need to order the runner que based on the DAG, but I don't think that will actually be too much work